crates/services/executor/src/executor.rs

```diff
@@ -1030,228 +1080,671 @@ where
+        Ok(())
+    }
+
+    fn check_mint_has_expected_index(
+        checked_mint: &Checked<Mint>,
+        execution_data: &ExecutionData,
+    ) -> ExecutorResult<()> {
+        if checked_mint.transaction().tx_pointer().tx_index() != execution_data.tx_count {
+            return Err(ExecutorError::MintHasUnexpectedIndex)
+        }
+        Ok(())
+    }
+
+    fn verify_mint_for_empty_contract(mint: &Mint) -> ExecutorResult<()> {
+        if *mint.mint_amount() != 0 {
+            return Err(ExecutorError::CoinbaseAmountMismatch)
+        }
+
+        let input = input::contract::Contract {
+            utxo_id: UtxoId::new(Bytes32::zeroed(), 0),
+            balance_root: Bytes32::zeroed(),
+            state_root: Bytes32::zeroed(),
+            tx_pointer: TxPointer::new(BlockHeight::new(0), 0),
+            contract_id: ContractId::zeroed(),
+        };
+        let output = output::contract::Contract {
+            input_index: 0,
+            balance_root: Bytes32::zeroed(),
+            state_root: Bytes32::zeroed(),
+        };
+        if mint.input_contract() != &input || mint.output_contract() != &output {
+            return Err(ExecutorError::MintMismatch)
+        }
+        Ok(())
+    }
+
+    fn store_mint_tx<T>(
+        mint: Mint,
+        execution_data: &mut ExecutionData,
+        coinbase_id: TxId,
+        block_st_transaction: &mut StorageTransaction<T>,
+    ) -> ExecutorResult<Transaction>
+    where
+        T: KeyValueInspect<Column = Column>,
+    {
+        let tx = mint.into();
+
+        execution_data.tx_status.push(TransactionExecutionStatus {
+            id: coinbase_id,
+            result: TransactionExecutionResult::Success {
+                result: None,
+                receipts: vec![],
+                total_gas: 0,
+                total_fee: 0,
+            },
+        });
+
+        if block_st_transaction
+            .storage::<ProcessedTransactions>()
+            .insert(&coinbase_id, &())?
+            .is_some()
+        {
+            return Err(ExecutorError::TransactionIdCollision(coinbase_id))
+        }
+        Ok(tx)
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn execute_chargeable_transaction<Tx, T>(
+        &self,
+        mut checked_tx: Checked<Tx>,
+        header: &PartialBlockHeader,
+        coinbase_contract_id: ContractId,
+        gas_price: Word,
+        execution_data: &mut ExecutionData,
+        tx_st_transaction: &mut StorageTransaction<T>,
+    ) -> ExecutorResult<Transaction>
+    where
+        Tx: ExecutableTransaction + PartialEq + Cacheable + Send + Sync + 'static,
+        <Tx as IntoChecked>::Metadata: CheckedMetadata,
+        T: KeyValueInspect<Column = Column>,
+    {
+        let tx_id = checked_tx.id();
+
+        if self.options.utxo_validation {
+            checked_tx = self.validate_utxos(checked_tx, header, tx_st_transaction)?;
+        }
+
+        let (reverted, state, mut tx, receipts) = self.attempt_tx_execution_with_vm(
```

Comment:
You need to `self.validate_inputs_state` before executing the transaction. User may lie to you about transaction's fields.

Reply:
As I mention above, before that was only for `Validation` before.

Reply:
I'm not sure if this addressed now that everything is moved into one function.